pkgs/by-name/ve/verapdf/package.nix

```diff
@@ -12,7 +12,17 @@ maven.buildMavenPackage rec {
   pname = "verapdf";
   version = "1.28.2";
 
-  mvnParameters = "-pl '!installer' -Dverapdf.timestamp=1980-01-01T00:00:02Z -Dproject.build.outputTimestamp=1980-01-01T00:00:02Z";
+  mvnParameters =
+    "-pl '!installer' -Dverapdf.timestamp=1980-01-01T00:00:02Z -Dproject.build.outputTimestamp=1980-01-01T00:00:02Z"
+    +
+      # By default, veraPDF uses version ranges for some components.
+      # These versions are pinned to the package version in order to avoid
+      # non-reproducibility of the maven dependencies.
+      lib.concatMapStrings (id: " -Dverapdf.${id}.version=${version}") [
+        "library"
+        "pdfbox.validation"
+        "validation"
+      ];
```

Comment:
Just a small nitpick: you could use `lib.concatMapStringsSep " "` instead of relying on a leading space in the string.